src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -70,9 +73,13 @@ public void deleteMessage(String ID) {
    */
   public List<Message> getMessagesbySubjectSearch(String searchCriteria) {
 
+    Filter searchFilter = new Query.FilterPredicate("subject", FilterOperator.EQUAL, searchCriteria);
+    Filter publicFilter = new Query.FilterPredicate("isPublic", FilterOperator.EQUAL, true);
```

Comment:
Nit: Since this public filter is used in several places and doesn't change, you could consider making it a class constant--so up on line 36 or so have a `private static final Filter PUBLIC_FILTER = ...`. (https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names for the all-caps naming advice)

Reply:
Ok that makes sense for the "is true" ones but what about when I make the DM page and its checking to see if its false? Should I make two static ones?